File: card-infra/src/main/java/com/jean202/cardmizer/infra/persistence/jpa/JpaSpendingRecordRepository.java
package com.jean202.cardmizer.infra.persistence.jpa;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface JpaSpendingRecordRepository extends JpaRepository<JpaSpendingRecordEntity, UUID> {
    List<JpaSpendingRecordEntity> findByDeletedFalseAndSpentOnBetweenOrderBySpentOnAscIdAsc(
            LocalDate startDate, LocalDate endDate);

    List<JpaSpendingRecordEntity> findByDeletedFalseAndSpentOnBetweenOrderBySpentOnDescIdDesc(
            LocalDate startDate, LocalDate endDate);

    List<JpaSpendingRecordEntity> findByDeletedFalseAndCardIdAndSpentOnBetweenOrderBySpentOnDescIdDesc(
            String cardId, LocalDate startDate, LocalDate endDate);
}

File: card-infra/src/main/kotlin/com/jean202/cardmizer/infra/persistence/jpa/JpaSpendingRecordRepository.kt
package com.jean202.cardmizer.infra.persistence.jpa

import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDate
import java.util.UUID

interface JpaSpendingRecordRepository : JpaRepository<JpaSpendingRecordEntity, UUID> {
    fun findByDeletedFalseAndSpentOnBetweenOrderBySpentOnAscIdAsc(
        startDate: LocalDate, endDate: LocalDate,
    ): List<JpaSpendingRecordEntity>

    fun findByDeletedFalseAndSpentOnBetweenOrderBySpentOnDescIdDesc(
        startDate: LocalDate, endDate: LocalDate,
    ): List<JpaSpendingRecordEntity>

    fun findByDeletedFalseAndCardIdAndSpentOnBetweenOrderBySpentOnDescIdDesc(
        cardId: String, startDate: LocalDate, endDate: LocalDate,
    ): List<JpaSpendingRecordEntity>
}
